enhancements/microshift/microshift-greenboot.md

```diff
@@ -0,0 +1,350 @@
+---
+title: microshift-greenboot
+authors:
+  - ggiguash
+reviewers:
+  - "@copejon"
+  - "@fzdarsky"
+  - "@majopela"
+  - "@dhellmann"
+  - "@oglok"
+  - "@zshi-redhat"
+  - "@pacevedom"
+  - "@runcom"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2022-12-22
+last-updated: 2022-12-22
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-319
+---
+
+# Integrating MicroShift with Greenboot
+
+## Summary
+
+Assure successful software upgrades and system configuration changes on Edge
+Devices running MicroShift by using [greenboot](https://github.com/fedora-iot/greenboot),
+the Generic Health Check Framework for `systemd` on `rpm-ostree` based systems.
+
+## Motivation
+
+Serviceability of Edge Devices is often limited or non-existent, which makes it
+challenging to troubleshoot device problems following a failed software or
+operating system upgrade.
+
+This functionality benefits the users by reducing the risk of being locked out
+of an Edge Device when upgrades take place. Users will not experience
+significant interruption of service in case of a failed upgrade.
+
+### User Stories
```

Comment:
The user stories below do not mention MicroShift at all, and so arguably in this generic form they could be considered solved today with greenboot and the generic health checks. Also, need to mention that they are optional. Something like:

"As a Device Administrator, I want the _option_ to include MicroShift into the greenboot health-checks, so that if MicroShift fails to start after an ostree update, the update is rolled back."

Reply:
Done